Fail
file path is :
projects/deltachat-core/rust/src/qr.rs

output is:


error is :
   Compiling deltachat v1.139.6 (/home/k/ogs_PHD/LLM4SE/2Rust_projects/auto_get_functionPair/projects/deltachat-core/rust)
error[E0425]: cannot find value `VCARD_BEGIN` in this scope
   --> src/qr.rs:272:30
    |
272 |     } else if qr.starts_with(VCARD_BEGIN) {
    |                              ^^^^^^^^^^^ not found in this scope

error[E0425]: cannot find value `DC_OPENPGP4FPR_SCHEME` in this scope
   --> src/qr.rs:277:30
    |
28  | const OPENPGP4FPR_SCHEME: &str = "OPENPGP4FPR:"; // yes: uppercase
    | ------------------------------------------------ similarly named constant `OPENPGP4FPR_SCHEME` defined here
...
277 |     } else if qr.starts_with(DC_OPENPGP4FPR_SCHEME) {
    |                              ^^^^^^^^^^^^^^^^^^^^^ help: a constant with a similar name exists: `OPENPGP4FPR_SCHEME`

error[E0425]: cannot find value `DC_OPENPGP4FPR_SCHEME` in this scope
   --> src/qr.rs:278:27
    |
28  | const OPENPGP4FPR_SCHEME: &str = "OPENPGP4FPR:"; // yes: uppercase
    | ------------------------------------------------ similarly named constant `OPENPGP4FPR_SCHEME` defined here
...
278 |         let payload = &qr[DC_OPENPGP4FPR_SCHEME.len()..];
    |                           ^^^^^^^^^^^^^^^^^^^^^ help: a constant with a similar name exists: `OPENPGP4FPR_SCHEME`

error[E0425]: cannot find function `decode_login` in this scope
   --> src/qr.rs:289:15
    |
289 |         match decode_login(context, qr).await {
    |               ^^^^^^^^^^^^ not found in this scope
    |
help: consider importing this function
    |
4   + use crate::qr::dclogin_scheme::decode_login;
    |

error[E0533]: expected value, found struct variant `Qr::Url`
   --> src/qr.rs:256:19
    |
256 |         return Ok(Qr::Url(qr.to_string()));
    |                   ^^^^^^^ not a value

error[E0425]: cannot find function `matmsg_to_qr` in this scope
   --> src/qr.rs:263:15
    |
263 |         match matmsg_to_qr(context, qr).await {
    |               ^^^^^^^^^^^^ not found in this scope

error[E0061]: this function takes 1 argument but 2 arguments were supplied
   --> src/qr.rs:284:15
    |
284 |         match decode_account(context, qr).await {
    |               ^^^^^^^^^^^^^^ ---------
    |                              |
    |                              unexpected argument of type `&context::Context`
    |                              help: remove the extra argument
    |
note: function defined here
   --> src/qr.rs:502:4
    |
502 | fn decode_account(qr: &str) -> Result<Qr> {
    |    ^^^^^^^^^^^^^^ --------

error[E0277]: `std::result::Result<Qr, anyhow::Error>` is not a future
   --> src/qr.rs:284:43
    |
284 |         match decode_account(context, qr).await {
    |               --------------------------- ^^^^^ `std::result::Result<Qr, anyhow::Error>` is not a future
    |               |
    |               this call returns `std::result::Result<Qr, anyhow::Error>`
    |
    = help: the trait `futures::Future` is not implemented for `std::result::Result<Qr, anyhow::Error>`, which is required by `std::result::Result<Qr, anyhow::Error>: std::future::IntoFuture`
    = note: std::result::Result<Qr, anyhow::Error> must be a future or must implement `IntoFuture` to be awaited
    = note: required for `std::result::Result<Qr, anyhow::Error>` to implement `std::future::IntoFuture`
help: remove the `.await`
    |
284 -         match decode_account(context, qr).await {
284 +         match decode_account(context, qr) {
    |
help: alternatively, consider making `fn decode_account` asynchronous
    |
502 | async fn decode_account(qr: &str) -> Result<Qr> {
    | +++++

error[E0425]: cannot find function `mailto_to_qr` in this scope
   --> src/qr.rs:294:15
    |
294 |         match mailto_to_qr(context, qr).await {
    |               ^^^^^^^^^^^^ not found in this scope

error[E0061]: this function takes 3 arguments but 2 arguments were supplied
   --> src/qr.rs:299:15
    |
299 |         match decode_ideltachat(context, qr).await {
    |               ^^^^^^^^^^^^^^^^^------------- an argument of type `&str` is missing
    |
note: function defined here
   --> src/qr.rs:493:10
    |
493 | async fn decode_ideltachat(context: &Context, prefix: &str, qr: &str) -> Result<Qr> {
    |          ^^^^^^^^^^^^^^^^^ -----------------  ------------  --------
help: provide the argument
    |
299 |         match decode_ideltachat(context, qr, /* &str */).await {
    |                                ~~~~~~~~~~~~~~~~~~~~~~~~~

error[E0533]: expected value, found struct variant `Qr::Text`
   --> src/qr.rs:305:8
    |
305 |     Ok(Qr::Text(qr.to_string()))
    |        ^^^^^^^^ not a value

Some errors have detailed explanations: E0061, E0277, E0425, E0533.
For more information about an error, try `rustc --explain E0061`.
error: could not compile `deltachat` (lib) due to 11 previous errors
warning: build failed, waiting for other jobs to finish...
error[E0277]: `std::result::Result<qr::Qr, anyhow::Error>` is not a future
   --> src/qr.rs:284:43
    |
284 |         match decode_account(context, qr).await {
    |               --------------------------- ^^^^^ `std::result::Result<qr::Qr, anyhow::Error>` is not a future
    |               |
    |               this call returns `std::result::Result<qr::Qr, anyhow::Error>`
    |
    = help: the trait `futures::Future` is not implemented for `std::result::Result<qr::Qr, anyhow::Error>`, which is required by `std::result::Result<qr::Qr, anyhow::Error>: std::future::IntoFuture`
    = note: std::result::Result<qr::Qr, anyhow::Error> must be a future or must implement `IntoFuture` to be awaited
    = note: required for `std::result::Result<qr::Qr, anyhow::Error>` to implement `std::future::IntoFuture`
help: remove the `.await`
    |
284 -         match decode_account(context, qr).await {
284 +         match decode_account(context, qr) {
    |
help: alternatively, consider making `fn decode_account` asynchronous
    |
502 | async fn decode_account(qr: &str) -> Result<Qr> {
    | +++++

error: could not compile `deltachat` (lib test) due to 11 previous errors
error: command `/home/k/.rustup/toolchains/1.77.1-x86_64-unknown-linux-gnu/bin/cargo test --no-run --message-format json-render-diagnostics` exited with code 101

